File: lib/models/Experience.dart
class Experience {
  final String role;
  final String company;
  final String startDate;
  final String endDate;
  final String description;

  Experience({
    required this.role,
    required this.company,
    required this.startDate,
    required this.endDate,
    required this.description,
  });

  factory Experience.fromJson(Map<String, dynamic> json) {
    return Experience(
      role: json['role'],
      company: json['company'],
      startDate: json['startDate'],
      endDate: json['endDate'],
      description: json['description'],
    );
  }
}

File: backend/src/main/kotlin/entityAndDataSourcesPortfolio/Experience.kt
package com.example.entityAndDataSourcesPortfolio

import kotlinx.serialization.Serializable
import org.bson.codecs.pojo.annotations.BsonId
import org.bson.types.ObjectId
//
//@Serializable
//data class Experience(
//    @BsonId  // for primary key of users
//    val id: String? = ObjectId().toString(),
//    val role: String,
//    val company: String,
//    val startDate: String,
//    val endDate: String?,
//    val description: String
//)

@Serializable
data class Experience(
    //@BsonId
    val id: String = ObjectId().toHexString(),  // CONVERT ObjectId to string
    val role: String,
    val company: String,
    val startDate: String,
    val endDate: String?,
    val description: String
)

